servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/apple/traffic/resilience/http/TrafficResilienceHttpServiceFilter.java

```diff
@@ -0,0 +1,688 @@
+         * seconds based on the duration the matching {@link CircuitBreaker} will remain open, or a fallback period.
+         */
+        public static BiConsumer<HttpResponseMetaData, StateContext>
+        retryAfterHintOfBreaker(final int fallbackSeconds) {
+            final CharSequence secondsSeq = newAsciiString(valueOf(fallbackSeconds));
+            return (resp, state) -> {
+                if (state.breaker() != null || fallbackSeconds > 0) {
+                    resp.setHeader(RETRY_AFTER, state.breaker() != null ? newAsciiString(valueOf(
+                                    state.breaker().remainingDurationInOpenState().getSeconds())) : secondsSeq);
+                }
+            };
+        }
+
+        /**
+         * A hard-coded delay in milliseconds to be supplied as a Retry-After-Millis HTTP header in a
+         * {@link HttpResponseMetaData}. Being a custom Http header, it will require special handling on the peer side.
+         *
+         * @param duration The duration to be used in the Retry-After-Millis header.
+         * @return A {@link HttpResponseMetaData} consumer, that enhances the headers with a fixed
+         * Retry-After-Millis figure in milliseconds.
+         */
+        public static BiConsumer<HttpResponseMetaData, CircuitBreaker> retryAfterMillisHint(final Duration duration) {
+            final CharSequence millisSeq = newAsciiString(valueOf(duration.toMillis()));
+            return (resp, breaker) -> resp.setHeader(RETRY_AFTER_MILLIS, millisSeq);
+        }
+
+        /**
+         * Pre-defined {@link StreamingHttpResponse response} that signals
+         * {@link io.servicetalk.http.api.HttpResponseStatus#TOO_MANY_REQUESTS} to the peer.
+         *
+         * @return A {@link BiFunction} that regardless the input, it will always return a
+         * {@link StreamingHttpResponseFactory#tooManyRequests() too-many-requests} response.
+         */
+        public static BiFunction<HttpRequestMetaData, StreamingHttpResponseFactory, Single<StreamingHttpResponse>>
+        tooManyRequests() {
+            return (__, factory) -> succeeded(factory.tooManyRequests());
+        }
+
+        /**
+         * Pre-defined {@link StreamingHttpResponse response} that signals
+         * {@link io.servicetalk.http.api.HttpResponseStatus#SERVICE_UNAVAILABLE} to the peer.
+         *
+         * @return A {@link BiFunction} that regardless the input, it will always return a
+         * {@link StreamingHttpResponseFactory#serviceUnavailable() service-unavailable} response.
+         */
+        public static BiFunction<HttpRequestMetaData, StreamingHttpResponseFactory, Single<StreamingHttpResponse>>
+        serviceUnavailable() {
+            return (__, factory) -> succeeded(factory.serviceUnavailable());
+        }
+
+        /**
+         * A {@link RejectionPolicy} builder to support a custom policy.
+         */
+        public static final class Builder {
```

Comment:
This will help to make the `Builder` more discoverable. Currently, there are too many inner class levels.